controllers/propagator/propagation.go

```diff
@@ -901,6 +937,83 @@ func (r *PolicyReconciler) processTemplates(
 			ManagedClusterName: decision.ClusterName,
 		}
 
+		// Handle value encryption initialization
+		if encryptionEnabled {
+			usesEncryption := templates.UsesEncryption(
+				policyT.ObjectDefinition.Raw, templateCfg.StartDelim, templateCfg.StopDelim,
+			)
+			// Initialize AES Key and initialization vector
+			if usesEncryption && !encryptionInitialized {
+				log.V(1).Info("Found an object definition requiring encryption. Handling encryption keys.")
+				// Get/generate the encryption key
+				encryptionKey, err := r.getEncryptionKey(decision.ClusterName)
+				if err != nil {
+					log.Error(err, "Failed to get AES key",
+						"policy", replicatedPlc.Name, "cluster", decision.ClusterName,
+					)
+				}
+
+				// Get/generate the initialization vector
+				initializationVector, newIV, err := r.getInitializationVector(
+					rootPlc.GetName(), decision.ClusterName, annotations,
+				)
+				if err != nil {
+					log.Error(err, "Failed to get initialization vector",
+						"policy", replicatedPlc.Name, "cluster", decision.ClusterName,
+					)
+				}
+
+				// If the initialization vector was generated, set it in the annotations
+				if newIV {
+					replicatedPlc.SetAnnotations(annotations)
+				}
+
+				// Set the EncryptionConfig with the retrieved key
+				if encryptionKey != nil && initializationVector != nil {
+					encryptionConfig := templates.EncryptionConfig{
+						EncryptionEnabled:    encryptionEnabled,
+						AESKey:               encryptionKey,
+						InitializationVector: initializationVector,
+					}
+
+					err = tmplResolver.SetEncryptionConfig(encryptionConfig)
+					if err != nil {
+						log.Error(
+							err, "Error setting encryption configuration",
+							"policy", replicatedPlc.Name, "cluster", decision.ClusterName,
+						)
+					}
+				}
+
+				encryptionInitialized = true
```

Comment:
Alternatively, you could set `templateCfg.EncryptionConfig` to `encryptionConfig` and then you could check against that being set or not instead of an extra boolean value.